File: Python/Two-Pointers/move-zeros.py
class Solution(object):
    def moveZeroes(self, nums):
        p1 = 0
        p2 = 0 
        while p2 < len(nums):
            if nums[p2] != 0:
                nums[p1],nums[p2] = nums[p2],nums[p1]
                p1 += 1
                p2 += 1
            else:
                p2 += 1


        print(nums)





x = Solution()
print(x.moveZeroes(nums=[0,1,0,3,12]))

File: Leet-Code/Two-pointers/movezeros.cc
#include <vector>

class Solution {
public:
  void moveZeroes(std::vector<int> &nums) {

    int r = 0;
    std::vector<int> nums1 = {};
    for (size_t i = 0; i < nums.size(); i++) {
      if (nums[i] != 0)
        nums1.push_back(nums[i]);
      if (nums[i] == 0)
        r++;
    }
    for (size_t i = 0; i < r; i++) {
      nums1.push_back(0);
    }
  }
};

int main() {

  Solution x;
  std::vector<int> nums1 = {0, 1, 0, 3, 12};
  x.moveZeroes(nums1);
}
